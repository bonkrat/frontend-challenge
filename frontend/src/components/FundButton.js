import React from "react";
import PropTypes from "prop-types";
import * as styles from "./FundButton.styles";

/**
 * Renders a button to fund a campaign/
 */
const FundButton = ({ loading, onClick }) => {
  return (
    <div
      className={styles.button(loading)}
      onClick={() => !loading && onClick()}
    >
      {loading ? "..." : "Fund"}
    </div>
  );
};

FundButton.defaultProps = {
  loading: false,
  onClick: () => {}
};

FundButton.propTypes = {
  /**
   * Whether the button should be inactive and loading.
   */
  loading: PropTypes.bool,
  /**
   * Handler for whent he button is clicked.
   */
  onClick: PropTypes.func
};

export default FundButton;
