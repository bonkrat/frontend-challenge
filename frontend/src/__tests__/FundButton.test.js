import React from "react";
import { shallow } from "enzyme";
import FundButton from "../components/FundButton";
import toJson from "enzyme-to-json";

let wrapper, mockFn;

beforeEach(() => {
  mockFn = jest.fn();

  const props = {
    loading: false,
    onClick: mockFn
  };

  wrapper = shallow(<FundButton {...props} />);
});

it("renders correctly", () => {
  expect(toJson(wrapper)).toMatchSnapshot();
});

it("renders loading styles", () => {
  wrapper.setProps({
    loading: true
  });
  wrapper.update();

  expect(toJson(wrapper)).toMatchSnapshot();
});

it("calls the onClick handler when not loading", () => {
  const cb = wrapper.prop("onClick");

  cb();

  expect(mockFn).toHaveBeenCalled();
});

it("does not call the onClick handler when not loading", () => {
  wrapper.setProps({
    loading: true
  });

  wrapper.update();
  const cb = wrapper.prop("onClick");

  cb();

  expect(mockFn).not.toHaveBeenCalled();
});
