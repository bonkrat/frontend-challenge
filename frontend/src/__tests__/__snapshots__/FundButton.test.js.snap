// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="rounded-lg border-2 border-white text-white py-2 px-4 ml-4 cursor-pointer bg-purple-700 hover:bg-purple-900"
  onClick={[Function]}
>
  Fund
</div>
`;

exports[`renders loading styles 1`] = `
<div
  className="rounded-lg border-2 border-white text-white py-2 px-4 ml-4 cursor-not-allowed hover:bg-gray-500 bg-gray-500 opacity-80"
  onClick={[Function]}
>
  ...
</div>
`;
